v3/onesignal-metabox/onesignal-metabox.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+defined('ABSPATH') or die('This page may not be accessed directly.');
+
+// Create a meta box
+add_action('add_meta_boxes', 'onesignal_add_metabox');
+
+function onesignal_add_metabox()
+{
+  add_meta_box(
+    'onesignal_metabox', // metabox ID
+    'OneSignal Push Notifications', // title
+    'onesignal_metabox', // callback function
+  );
+}
+
+// Render the meta box
+function onesignal_metabox($post)
+{
+  // View segments API to populate Segments list.
+  $args = array(
+    'headers' => array(
+      'Authorization' => 'Bearer ' . get_option('OneSignalWPSetting')['app_rest_api_key'],
+      'accept' => 'application/json',
+      'content-type' => 'application/json',
+    )
+  );
+
+  // Make the API request, log errors, get segment names.
+  $response = wp_remote_get('https://onesignal.com/api/v1/apps/' . get_option('OneSignalWPSetting')['app_id'] . '/segments', $args);
+  if (is_wp_error($response)) {
+    error_log('API request failed: ' . $response->get_error_message());
+  }
+  $json = json_decode(wp_remote_retrieve_body($response));
```

Comment:
Is this going to break if $response is an error?

Reply:
It won't break. It'll fail silently